src/core/queryCache.js

```diff
@@ -8,16 +8,16 @@ import {
 import { defaultConfigRef } from './config'
 import { makeQuery } from './query'
 
-export const queryCache = makeQueryCache()
+export const queryCache = makeQueryCache({ frozen: isServer })
```

Comment:
Hmm. I had someone trying to use ReactQuery in the CLI the other day. This would make that default to frozen, correct?

Reply:
Yes, the global cache will still be frozen by default in all server environments in this PR since changing that behavior would regress #70 and in my eyes be a breaking change.

I would suggest not only that we change this default behavior in the next major, but actually that we get rid of the global cache entirely (at least on the server side, possibly client too). So instead of making the global cache available for CLI-development, we push _everyone_ to create their own cache.

Creating a `queryCache` via `makeQueryCache` is trivial and you can easily just export that as a global cache yourself if that is what you want. What we would gain from this is that developers would explicitly have to create a cache somewhere, making it clear that whatever you fetch into this cache will be shared. This would let us get rid of the frozen/unfrozen concept, streamline the behavior and still make it hard to/avoid accidentally leaking data cross-request.

Reply:
While I totally wish I had avoided the singleton from the start, it's going to be very difficult to get away from in the future. Even IMO, it's very convenient. I'd like to avoid it as long as possible ;)